### DISCUSSION TITLE
[Feature Request]: after rows (data) load hook

### DISCUSSION THREAD

**Discussion by xabec** [rappasoft/laravel-livewire-tables#1487] [Feature Requests]
### Overview

Hi, at first, thank you for your hard work on this great package. It would be awesome if it would have some kind of hooks or ways to alter data before loading it. I was trying to override getRows() method but yet the data is manipulated somehow that I can't recreate data format no matter what I try and it's not very convenient either way. It would be awesome if we could adjust some things before data is loaded and after data is loaded. For example change some values, add extra custom row or whatever. 

If you have any further questions what do I mean let me know. 

### Detailed explanation

When I add F.E. into my DataTableComponent

```
public function afterDataLoadHook($data) {
$data[] = [
'date' => '2023-10-27',
'pick' => '1',
'tries' => '5'
];

return $data;
}
```

or whatever it would add a new row to existing data before rendering the table.

**Comment by lrljoe** [comment]
Please reach out on the Official Discord to discuss this further, as I'm missing the purpose/use-case here!

**Comment by lrljoe** [comment]
There are some hooks planned (configuring/configured for example), but don't know why you'd want to call at that point being honest.  Happy to consider adding hooks anywhere, but need to understand the use case really!

**Reply by lrljoe** [reply]
So only problem with this at the moment, is getRows is called at render, as it always needs to run,  So I'll need to look at moving it somewhere else potentially, or calling the hooks within getRows()

Might be easier to add an append/prepend row feature to be honest to make it more standardised
